scripts/zones/Misareaux_Coast/mobs/Alsha.lua

```diff
@@ -0,0 +1,50 @@
+-----------------------------------
+--  Area: Misareaux Coast
+--    NM: Alsha
+-- Quest: Knocking on Forbidden Doors
+-----------------------------------
+local ID = require("scripts/zones/Misareaux_Coast/IDs")
+require("scripts/globals/status")
+-----------------------------------
+local entity = {}
+
+entity.onMobSpawn = function(mob)
+    mob:setLocalVar("CureTime", os.time() + math.random(60, 180)) -- anywhere from every 1-3 minutes
+end
+
+entity.onMobFight = function(mob, target)
+    -- Check to see if below 20% HPP, and set the "final mode" active
+    if mob:getLocalVar("FinalPhase") ~= 1 and mob:getHPP() < 20 then
+        target:messageSpecial(ID.text.LARGE_DROPS_OF_OIL)
+        mob:setLocalVar("FinalPhase", 1)
+        -- TODO: Uncomment these lines
+        -- Fix code below so that Alsha will cast Fire II on herself.  Currently not possible without core changes.
```

Comment:
I have been absolutely stumped on this kind of behavior for a while. We should work together sometime and try to figure it out!

Reply:
For sure.  It will need some core changes.  I'd like to build in an override on the target (i.e. by default, you can't cast Fire on yourself or an ally, but under certain conditions, we need to allow it).  Just gotta figure that out.  Can do that this week.

FYI, this entire PR is moved up to LSB, so I'm putting it on hold until I get it merged.  Keeping it open here just in case.

Reply:
money man. I was thinking we could customize the lua call in some shape or form. This way it'll only be allowed if it's scripted to do so in lua. Perhaps even have the lua call an entirely new function to handle the cast.